Gravatar/MruImageCache.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Gravatar
+{
+    /// <summary>
+    /// Caches most-recently-used images.
+    /// </summary>
+    /// <remarks>
+    /// Decorates an inner cache, delegating to it as needed.
+    /// <para />
+    /// If an image is available in memory, the inner cache can be bypassed.
+    /// </remarks>
+    public sealed class MruImageCache : IImageCache
+    {
+        private const int CleanAtSize = 30;
+        private const int CleanToSize = 25;
+
+        private readonly Dictionary<string, Entry> _entryByFileName = new Dictionary<string, Entry>();
```

Comment:
Suggest `ConcurrentDictionary` as it is thread-safe.

Reply:
Sounds good. Pushed.

The algorithm in `RemoveOldEntries` might potentially remove too many items if run concurrently, but that doesn't really matter.

In practice I assume this will only be used on the UI thread anyway.